eng/WpfArcadeSdk/tools/InjectModuleInitializer.targets

```diff
@@ -0,0 +1,423 @@
+    <PropertyGroup>
+      <DotNetFrameworkBitness Condition="'$(Platform)'!='AnyCPU' and '$(Platform)'!='Win32'">Bitness64</DotNetFrameworkBitness>
+      <DotNetFrameworkBitness Condition="'$(Platform)'=='AnyCPU' or '$(Platform)'=='Win32'">Bitness32</DotNetFrameworkBitness>
+    </PropertyGroup>
+    <FindDotNetFrameworkILToolsTask DotNetBitness="$(DotNetFrameworkBitness)">
+      <Output TaskParameter="ILDAsm" PropertyName="ILDAsm" />
+      <Output TaskParameter="ILAsm" PropertyName="ILAsm" />
+    </FindDotNetFrameworkILToolsTask>
+  </Target>
+
+  <!-- Disassemble the target binary. -->
+  <Target Name="DeconstructDll"
+        AfterTargets="CreateSatelliteAssemblies"
+        Inputs="$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName)$(TargetExt)'))"
+        Outputs="$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName).il'))">
+    <PropertyGroup>
+      <InFile>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName)$(TargetExt)'))</InFile>
+      <OutFile>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName).il'))</OutFile>
+    </PropertyGroup>
+
+    <Delete Files="$(OutFile)"
+            Condition="Exists('$(OutFile)')" />
+
+    <ILDasmTask ILDAsm="$(ILDAsm)"
+                Assembly="$(InFile)"
+                Condition="Exists('$(InFile)')"
+                LineNum="true"
+                Out="$(OutFile)" />
+
+  </Target>
+
+  <!-- Inject the module initializer into the target IL. -->
+  <Target Name="AddModuleConstructor"
+          AfterTargets="DeconstructDll">
+    <PropertyGroup>
+      <ILFile>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName).il'))</ILFile>
+      <ModuleInitializer>$(WpfArcadeSdkToolsDir)InjectModuleInitializer\ModuleInitializer.il</ModuleInitializer>
+      <MsCorLibAssemblySection>$(WpfArcadeSdkToolsDir)InjectModuleInitializer\MsCorLibAssemblySection.il</MsCorLibAssemblySection>
+    </PropertyGroup>
+
+    <AddModuleConstructorTask ILFile="$(ILFile)"
+                              MsCorLibAssemblySectionIL="$(ModuleInitializer)"
+                              ModuleConstructorIL="$(MsCorLibAssemblySection)"/>
+
+  </Target>
+
+  <!-- Re-assembly the target binary. -->
+  <Target Name="ReconstituteDll"
+          AfterTargets="AddModuleConstructor"
+          Inputs="$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName).il'))"
+          Outputs="$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName)$(TargetExt)'))">
+    <PropertyGroup>
+      <TargetFile>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName)$(TargetExt)'))</TargetFile>
+      <ILFile>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(AssemblyName).il'))</ILFile>
+    </PropertyGroup>
+
+    <!-- 
+      Make a backup before overwriting $(TargetFile) 
+    -->
+    <Copy SourceFiles="$(TargetFile)"
```

Comment:
Should we decompile again and diff =against the previous IL file to check validity?

Reply:
I diffed round tripped IL by straight text diff and it seems to be a very noisy affair.  I am not sure we can easily do this, but I'll look around for something to make this possible.

Reply:
After looking around a bit, the alignment changes from roundtripping (and just the general size of the diff) makes this prohibitive during build.  I did some spot checks on 10 or so functions and it all seemed to be the same.  I think we can just say forget roundtrip diffs for now.